widget/hexwidget/hexwidget.go

```diff
@@ -0,0 +1,214 @@
+}
+
+func (h *hexRenderer) Layout(size fyne.Size) {
+}
+
+func (h *hexRenderer) ApplyTheme() {
+}
+
+func (h *hexRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (h *hexRenderer) Refresh() {
+	for i, v := range h.segmentObjects {
+		v.(*canvas.Line).StrokeColor = h.hex.getSegmentColor(i)
+		canvas.Refresh(v)
+	}
+}
+
+func (h *hexRenderer) Destroy() {
+}
+
+func (h *hexRenderer) Objects() []fyne.CanvasObject {
+	return h.segmentObjects
+}
+
+// HexWidget represents a 7-segment hexadecimal display. The segments
+// of the display mapped active-low onto 7 state bits, with segment 0 in
+// the least significant bit.
+//
+//       0
+//     -----
+//    |     |
+//  5 |     | 1
+//    |  6  |
+//     -----
+//    |     |
+//  4 |     | 2
+//    |  3  |
+//     -----
+type HexWidget struct {
+	widget.BaseWidget
+	segments uint8
+}
+
+func ptToPos(pt image.Point) fyne.Position {
+	return fyne.NewPos(float32(pt.X), float32(pt.Y))
+}
+
+func setLineEndpoints(l *canvas.Line, pt1, pt2 image.Point) {
+	l.Move(fyne.NewPos(float32(pt1.X), float32(pt1.Y)))
+	l.Resize(fyne.NewSize(float32(pt2.X-pt1.X), float32(pt2.Y-pt1.Y)))
+}
+
+func (h *HexWidget) getSegmentColor(segno int) color.RGBA {
+	if (h.segments & (1 << segno)) == 0 {
+		return hexOnColor
+	}
+
+	return hexOffColor
+}
+
+// CreateRenderer implements fyne.Widget
+func (h *HexWidget) CreateRenderer() fyne.WidgetRenderer {
+	seg0 := canvas.NewLine(hexOffColor)
+	seg1 := canvas.NewLine(hexOffColor)
+	seg2 := canvas.NewLine(hexOffColor)
+	seg3 := canvas.NewLine(hexOffColor)
+	seg4 := canvas.NewLine(hexOffColor)
+	seg5 := canvas.NewLine(hexOffColor)
+	seg6 := canvas.NewLine(hexOffColor)
+
+	seg0.StrokeWidth = float32(hexSegmentWidth / 2)
+	seg1.StrokeWidth = float32(hexSegmentWidth / 2)
+	seg2.StrokeWidth = float32(hexSegmentWidth / 2)
+	seg3.StrokeWidth = float32(hexSegmentWidth / 2)
+	seg4.StrokeWidth = float32(hexSegmentWidth / 2)
+	seg5.StrokeWidth = float32(hexSegmentWidth / 2)
+	seg6.StrokeWidth = float32(hexSegmentWidth / 2)
+
+	pos := image.Pt(0, 0)
+
+	// pt0Center := image.Pt(pos.X+int(hexWidth/2.0+hexOffset), pos.Y+int((hexHeight-2.0*hexSegmentVLength)/2))
+	pt0Center := image.Pt(pos.X+int(hexWidth/2.0+hexOffset), pos.Y)
+	pt05 := image.Pt(int(float32(pt0Center.X)-(hexSegmentHLength/2)), pt0Center.Y)
+	pt01 := image.Pt(int(float32(pt0Center.X)+(hexSegmentHLength/2)), pt0Center.Y)
+
+	pt6Center := image.Pt(pos.X+int(hexWidth/2.0), int(float32(pt0Center.Y)+hexSegmentVLength))
+	pt65 := image.Pt(int(float32(pt6Center.X)-(hexSegmentHLength/2)), pt6Center.Y)
+	pt61 := image.Pt(int(float32(pt6Center.X)+(hexSegmentHLength/2)), pt6Center.Y)
+
+	pt3Center := image.Pt(pos.X+int(hexWidth/2.0-hexOffset), int(float32(pt0Center.Y)+2*hexSegmentVLength))
+	pt34 := image.Pt(int(float32(pt3Center.X)-(hexSegmentHLength/2)), pt3Center.Y)
+	pt32 := image.Pt(int(float32(pt3Center.X)+(hexSegmentHLength/2)), pt3Center.Y)
+
+	// segment 0
```

Comment:
I think either this comment is not needed, or the `seg0` should be `segment0` if you worry it's not clear

Reply:
df17679c1ae6d6f2f3ad755f0ab930a11340f923